src/SessionMonitor/index.js

```diff
@@ -83,6 +83,9 @@ export class SessionMonitor {
         if (response.type === 'UPDATE_POPUP') {
           this.popupShown = true;
         }
+      }).catch(() => {
+        // if API failed check if user is still logged in
+        this.notifyUserIfTheyAreNotLoggedIn();
```

Comment:
why logging out user at here? the logic was if it can reach `refreshSession()` then it means the user is still logged in and actively using the portal

Reply:
the bug had to do with the user being logged out, but not alerted that they had been logged out, leading to other errors. this code checks if they are still logged in and if not alerts them, it does not log them out. I am still not 100% on how all of this code works if you do not think this check is needed i can remove

Reply:
I think the bug was actually buried deeper in the logic flow: the `notifyUserIfTheyAreNotLoggedIn` function will make a call to the submission api and when it returns a 401 the portal assumes the user is logged out. But this qill result in as the current behavior that for some times the popup showed up early while in other occassions some are shown up later.

This is mainly due to the fact that the access token in cookie is not updated until the pervious one has expired. After this fix https://github.com/uc-cdis/fence/pull/874 is merged, both session token and access token will be updated.

For that case, I probably will test and change this logic flow back to let portal actively logout users. So this piece of code will probably be removed in my work. I'm okay will merging as is because I think it won't break things, just a FYI

The changes to the workspace code LGTM

Reply:
sounds good, thanks for looking over, im glad someone else was also looking into this